solution_collector/src/main.rs

```diff
@@ -12,31 +14,104 @@ use github_rs::client::{Github, Executor};
 use serde_json::Value;
 use regex::Regex;
 use std::collections::HashMap;
+use structopt::StructOpt;
 
-#[derive(Debug)]
-enum Mode {
+#[derive(StructOpt, Debug)]
+#[structopt(name = "solution_collector")]
+struct Opt {
+    #[structopt(subcommand)]
+    command: Command
+}
+
+#[derive(StructOpt, Debug)]
+enum Command {
+    #[structopt(name = "preview")]
     Preview,
-    Merge
+    #[structopt(name = "week-report")]
+    WeekReport {
+        week: u32
+    },
+    #[structopt(name = "merge")]
+    Merge {
+        week: u32,
+    }
+}
+
+struct Context {
+    owner: String,
+    repo: String,
+    client: Github
+}
+
+impl Context {
+    fn new(token: String, owner: String, repo: String) -> Self {
+        let client = Github::new(token).expect("Invalid token");
+        Context {
+            owner,
+            repo,
+            client
+        }
+    }
+}
+
+#[derive(Debug)]
+struct Problem {
+    title: String,
+    done: bool
+}
+
+#[derive(Debug)]
+struct SubmitEntry {
```

Comment:
@unrealhoang Could you add a field for submitted date, and using this field for checking a PR is submitted after the deadline?

Reply:
nice catch, will add it.